src/lib/util.ts

```diff
@@ -24,3 +24,22 @@ export function bind(target: any, propertyKey: string, descriptor: PropertyDescr
     }
   };
 }
+
+/**
+ * Turns a given `ImageBitmap` into `ImageData`.
+ */
+export async function bitmapToImageData(bitmap: ImageBitmap): Promise<ImageData> {
+  // Make canvas same size as image
+  // TODO: Move this off-thread if possible with `OffscreenCanvas` or iFrames?
```

Comment:
We could just cache this for now so it only runs when the source image changes. OffscreenCanvas is still behind a flag and iframe would have to be served from a different origin to have any benefit :(